clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -172,29 +189,6 @@ public synchronized boolean updateLastSeenEpochIfNewer(TopicPartition topicParti
         return Optional.ofNullable(lastSeenLeaderEpochs.get(topicPartition));
     }
 
-    /**
-     * Conditionally update the leader epoch for a partition
-     *
-     * @param topicPartition       topic+partition to update the epoch for
-     * @param epoch                the new epoch
-     * @param epochTest            a predicate to determine if the old epoch should be replaced
-     * @return true if the epoch was updated, false otherwise
-     */
-    private synchronized boolean updateLastSeenEpoch(TopicPartition topicPartition,
```

Comment:
I removed this since at this point the two callers have different needs for updating the epoch